src/rust/ide/lib/parser/src/jsclient.rs

```diff
@@ -78,4 +81,12 @@ impl Client {
         };
         Ok(result()?)
     }
+
+    pub fn doc_parser_generate_html_source(&self, program:String) -> api::Result<String> {
+        let html_code = || {
+            let html_code = doc_parser_generate_html_source(program)?;
+            Result::Ok(html_code)
+        };
+        Ok(html_code()?)
+    }
```

Comment:
1. I would name this function just "generate_html_docs"
2. Add the docstring to this function.
3. Cannot the lambda be just the function body?

Reply:
About all renames - this is the same name as there is in `parser-service` on the engine side and in `scala-parser.js`, that's why i used it, so it all feels unified